File: loop_transformations/sources/apply_syntax.1.c
/*
* @@name:	apply_syntax.1
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
void construct_unroll(double A[100])
{
   #pragma omp unroll
   #pragma omp tile sizes(4)
   for (int i = 0; i < 100; ++i)
      A[i] = A[i] + 1;
}

void apply_unroll(double A[100])
{
   #pragma omp tile sizes(4) apply(grid: unroll)
   for (int i = 0; i < 100; ++i)
      A[i] = A[i] + 1;
}

File: loop_transformations/sources/apply_syntax.1.f90
! @@name:       apply_syntax.1
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0
subroutine construct_unroll(A)
   implicit none
   integer :: i
   double precision :: A(0:99)

   !$omp unroll
   !$omp tile sizes(4)
   do i = 0, 99
      A(i) = A(i) + 1
   end do
end subroutine

subroutine apply_unroll(A)
   implicit none
   integer :: i
   double precision :: A(0:99)

   !$omp tile sizes(4) apply(grid: unroll)
   do i = 0, 99
      A(i) = A(i) + 1
   end do
end subroutine
